fineract-provider/src/main/java/org/apache/fineract/portfolio/client/mapper/ClientMapper.java

```diff
@@ -0,0 +1,159 @@
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.client.mapper;
+
+import org.apache.fineract.infrastructure.codes.data.CodeValueData;
+import org.apache.fineract.infrastructure.codes.domain.CodeValue;
+import org.apache.fineract.infrastructure.core.config.MapstructMapperConfig;
+import org.apache.fineract.infrastructure.core.data.EnumOptionData;
+import org.apache.fineract.portfolio.client.data.ClientData;
+import org.apache.fineract.portfolio.client.data.ClientTimelineData;
+import org.apache.fineract.portfolio.client.domain.Client;
+import org.apache.fineract.portfolio.client.domain.ClientEnumerations;
+import org.apache.fineract.useradministration.domain.AppUser;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+@Mapper(config = MapstructMapperConfig.class)
+public interface ClientMapper {
+
+    @Mapping(target = "accountNo", source = "source.accountNumber")
+    @Mapping(target = "status", source = "source", qualifiedByName = "clientStatusEnum")
+    @Mapping(target = "subStatus", source = "source", qualifiedByName = "clientSubStatusCode")
+    @Mapping(target = "officeId", source = "source.office.id")
+    @Mapping(target = "officeName", source = "source.office.name")
+    @Mapping(target = "transferToOfficeId", source = "source.transferToOffice.id")
+    @Mapping(target = "transferToOfficeName", source = "source.transferToOffice.name")
+    @Mapping(target = "externalId", source = "source.externalId")
+    @Mapping(target = "gender", source = "source", qualifiedByName = "clientGenderCode")
+    @Mapping(target = "imageId", source = "source.image.id")
+    @Mapping(target = "staffId", source = "source.staff.id")
+    @Mapping(target = "staffName", source = "source.staff.displayName")
+    @Mapping(target = "timeline", source = "source", qualifiedByName = "clientTimeloneData")
+    @Mapping(target = "savingsProductId", source = "source.savingsProductId")
+    @Mapping(target = "savingsProductName", source = "source.id")
+    @Mapping(target = "savingsAccountId", source = "source.savingsAccountId")
+    @Mapping(target = "clientType", source = "source", qualifiedByName = "clientTypeCode")
+    @Mapping(target = "clientClassification", source = "source", qualifiedByName = "clientClassificationCode")
+    @Mapping(target = "legalForm", source = "source", qualifiedByName = "clientLegalFormEnum")
+    @Mapping(target = "isStaff", source = "source", qualifiedByName = "clientIsStaff")
+    @Mapping(target = "imagePresent", ignore = true)
```

Comment:
So why do we ignore these props if they are present in the ClientData class?

Reply:
The `ClientData` dto is used for different scenarios, Individual clients, Group clients for instance. So In this specific case we are talking just for Individual clients and the other attributes are not filled/used in this case and that is the way to be not filled (ignoring them)